### PR TITLE
Issue #3 | Create DockerFile / docker-compose final

### PR DESCRIPTION
**Pull request by icep0ps**
I just added the dockerfile with a basic setup for when we need to add next as a service in the compose file

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+version: '3.8'
+services:
+  db:
+    container_name: postgres
+    image: postgis/postgis:15-3.3
```
Comment (BenocxX) [inline docker-compose.yaml:5]: Change image for default Postgres image. Should be something like: `image: postgres:14-bullseye`

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+version: '3.8'
+services:
+  db:
+    container_name: postgres
```
Comment (BenocxX) [inline docker-compose.yaml:4]: Change name for `code-racer-postgres`. That way, we won't confuse the db with one from an other project. Clarity purposes but not essential

File: Dockerfile
Comment (BenocxX) [inline Dockerfile:1]: Remove this file since we won't be using Docker for production yet.

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    image: postgis/postgis:15-3.3
+    restart: always
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
+    ports:
+      - '5432:5432'
+    volumes:
```
Comment (BenocxX) [inline docker-compose.yaml:12]: Add a volume for `init.sql` files.

Should be something like that:
`- ./sql/init.sql:/docker-entrypoint-initdb.d/init.sql`
 
Will need to be tested locally to make sure it's working. Basically, we want the SQL written in `init.sql` to be automatically executed on db initialization. What I mean by that is that when we spin up a new instance of the db, the code in `init.sql` should be ran to populate the db.

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+version: '3.8'
+services:
+  db:
+    container_name: postgres
+    image: postgis/postgis:15-3.3
+    restart: always
+    environment:
+      - POSTGRES_USER=postgres
```
Comment (BenocxX) [inline docker-compose.yaml:8]: If you have a `.env` file, you should also put the user name in it.

File: docker-compose.yaml (hunk 3 above)
Comment (BenocxX) [reply docker-compose.yaml:12]: Forget this part! I think prisma offers a seeding system.. We'll need to be investigated!

File: docker-compose.yaml (hunk 4 above)
Comment (sudo-adduser-jordan) [reply docker-compose.yaml:8]: Check .env.example on the main branch. You may have forked the respository before it was merged. You can load the .env variables directly like this: 
```
    env_file:
      - .env
    ports:
      - ${DB_PORT}:${DB_PORT}
    environment:
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
      - POSTGRES_DB=${DB_NAME}
```

**Review by sudo-adduser-jordan — CHANGES_REQUESTED** [review]
Update
```
image: postgres
```

Delete
```
- ./sql/init.sql:/docker-entrypoint-initdb.d/init.sql
driver: local
```
I don't think we need the init.sql or driver as prisma will provide those features.

**Comment by icep0ps** [comment]
> Update
> 
> ```
> image: postgres
> ```
> 
> Delete
> 
> ```
> - ./sql/init.sql:/docker-entrypoint-initdb.d/init.sql
> driver: local
> ```
> 
> I don't think the init.sql or driver is needed as prisma will provide those features.

dont we need to lock the imge version because image: postgres will just pull the latest version

**Comment by BenocxX** [comment]
Looks good to me!

**Comment by sudo-adduser-jordan** [comment]
Yes, it should be locked. Bookworm is the latest stable release of debian.
```
image: postgres:15.3-bookworm
```

**Pull request merged**